applications/luci-app-sshtunnel/htdocs/luci-static/resources/view/sshtunnel.js

```diff
@@ -0,0 +1,403 @@
+		s.addremove = true;
+		s.nodescriptions = true;
+
+		o = s.option(form.Flag, "enabled", _("Enabled"));
+		o.default = "1";
+
+		o = _addServerOption(s);
+
+		o = s.option(form.ListValue, "vpntype", _("VPN type"), "");
+		o.value("point-to-point", "TUN (point-to-point)");
+		o.value("ethernet", "TAP (ethernet)");
+		o.default = "point-to-point";
+
+		o = s.option(form.Value, "localdev", _("Local dev"), "");
+		o.default = "any";
+		o.datatype = 'or("any", min(0))';
+		o.rmempty = false;
+
+		o = s.option(form.Value, "remotedev", _("Remote dev"), "");
+		o.default = "any";
+		o.datatype = 'or("any", min(0))';
+		o.rmempty = false;
+
+
+		s = m.section(form.GridSection, "_keys");
+		s.render = L.bind(_renderSshKeys, this, sshKeys);
+
+		s = m.section(form.GridSection, "_known_hosts");
+		s.render = L.bind(_renderKnownHosts, this, knownHosts);
+
+		return m.render();
+	},
+});
+
+function _addServerOption(s) {
+	var o = s.option(form.ListValue, "server", _("Server"), _("Type a server name from servers list above"));
+	o.datatype = "uciname";
+	o.rmempty = false;
+	uci.sections('sshtunnel', 'server', function (s, sectionName) {
+		o.value(sectionName, s.hostname ? '%s (%s)'.format(sectionName, s.hostname) : sectionName);
+	});
+	return o;
+}
+
+function _splitSshKeys(sshFiles) {
+	var sshKeys = {};
+	for (var i = 0; i < sshFiles.length; i++) {
+		var sshPubKeyName = sshFiles[i];
+		var sshKeyName = sshPubKeyName.substring(0, sshPubKeyName.length - 4);
+		i++;
+		var sshPub = sshFiles[i];
+		sshKeys[sshKeyName] = "<small><code>" + sshPub + "</code></small>";
+	}
+	allSshKeys = sshKeys;
+	return sshKeys;
+}
+
+function _renderSshKeys(sshKeys) {
+	var table = E('table', {'class': 'table cbi-section-table', 'id': 'keys_table'}, [
+		E('tr', {'class': 'tr table-titles'}, [
+			E('th', {'class': 'th'}, _('Name')),
+			E('th', {'class': 'th'}, _('Public Key')),
+		])
+	]);
+
+	var rows = Object.entries(sshKeys);
+	cbi_update_table(table, rows, E('em', _("No any SSH key, please generate a new one")));
+
+	var keyGenBtn = E('div', {}, [
+		E('label', {}, _('Generate a new key') + ': '),
+		E('span', {'class': 'control-group'}, [
+			E('input', {
+				'type': 'text',
+				'name': 'keyName',
+				'value': 'id_ed25519',
+				'keydown': function (ev) {
+					if (ev.keyCode === 13) _handleKeyGen(ev);
+				}
+			}),
+			E('button', {
+				'class': 'btn cbi-button cbi-button-action',
+				'click': _handleKeyGen
+			}, [_('Generate')])
+		])
+	]);
+	return E('div', {'class': 'cbi-section cbi-tblsection'}, [
+		E('h3', _('Keys')),
+		E('div', {'class': 'cbi-section-descr'},
+			_("You need to add the pub key to /root/.ssh/authorized_keys or to /etc/dropbear/authorized_keys.") + " " +
+			_("In LuCI you can do that with System / Administration / SSH-Keys")
+		),
+		keyGenBtn, table
+	]);
+}
+
+function _handleKeyGen(event) {
+	var keyName = document.querySelector('input[name="keyName"]').value;
+	if (!keyName) {
+		alert(_("Please specify a key name"));
```

Comment:
`return _("Specify a key name.");`

DO NOT USE `alert()`.

Reply:
This is an onclick event so we can't return anything. What's wrong with the alert()?

Reply:
changed to uiNotification